core/cache/lxdprofilewatcher.go

```diff
@@ -98,20 +98,21 @@ func (w *MachineAppLXDProfileWatcher) applicationCharmURLChange(topic string, va
 			w.logError(fmt.Sprintf("error getting charm %s to evaluate for lxd profile notification: %s", chURL, err))
 			return
 		}
+
 		// notify if:
 		// 1. the prior charm had a profile and the new one does not.
 		// 2. the new profile is not empty.
-		if (info.charmProfile != nil && ch.details.LXDProfile.Empty()) ||
-			!ch.details.LXDProfile.Empty() {
+		lxdProfile := ch.LXDProfile()
+		if (!info.charmProfile.Empty() && lxdProfile.Empty()) || !lxdProfile.Empty() {
```

Comment:
I definitely prefer Empty vs Empty to 'nil vs Empty'.
Is there a different benefit for not using a pointer?